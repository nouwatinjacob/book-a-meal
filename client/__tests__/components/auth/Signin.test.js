import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  Login,
  mapStateToProps
} from '../../../src/components/auth/Signin.jsx';

const props = {
  loginAction: () => Promise.resolve(),
  isLoading: true
};

describe('Signin component', () => {
  const setup = () => shallow(<Login {...props} />);

  it('should render Signin component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'onSubmit'
  when signin button is clicked`, (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    };

    wrapper.setState({
      loginData: {
        email: 'jaysansa@gmail.com',
        password: 'hghghghgh'
      }
    });

    const onSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'onSubmit'
    );
    wrapper.instance().onSubmit(event);
    expect(onSubmitSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onInputChange'
  when input value is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        name: {
          email: 'jaysansa@gmail.com',
          password: 'hghghghgh'
        }
      }
    };

    const onInputChangeSpy = jest.spyOn(
      wrapper.instance(),
      'onInputChange'
    );
    wrapper.instance().onInputChange(event);
    expect(onInputChangeSpy).toHaveBeenCalled();

    done();
  });
});