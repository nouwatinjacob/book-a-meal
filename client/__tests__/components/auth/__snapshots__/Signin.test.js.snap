// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin component should render Signin component correctly 1`] = `
<div
  className="container"
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <div
      className="login"
    >
      <div
        className="login-form"
      >
        <Loader
          color="#ff9600"
          height="50"
          margin="2px"
          type="Rings"
          visible={true}
          width="100"
        />
        <h3>
          User Login
        </h3>
        <form
          onSubmit={[Function]}
        >
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
          <br />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <br />
          <button
            className="button warning"
          >
            Login
          </button>
          <p>
            Forgot 
            <a
              href="#"
            >
              Password?
            </a>
          </p>
        </form>
      </div>
    </div>
  </div>
</div>
`;
